scipy/stats/_continuous_distns.py

```diff
@@ -3727,12 +3727,17 @@ def _rvs(self, mu, size=None, random_state=None):
         return random_state.wald(mu, 1.0, size=size)
 
     def _pdf(self, x, mu):
-        # invgauss.pdf(x, mu) =
-        #                  1 / sqrt(2*pi*x**3) * exp(-(x-mu)**2/(2*x*mu**2))
-        return 1.0/np.sqrt(2*np.pi*x**3.0)*np.exp(-1.0/(2*x)*((x-mu)/mu)**2)
+        return np.exp(self._logpdf(x, mu))
 
     def _logpdf(self, x, mu):
-        return -0.5*np.log(2*np.pi) - 1.5*np.log(x) - ((x-mu)/mu)**2/(2*x)
+        a = -0.5 * np.log(2 * np.pi)
+        out = _lazywhere(
+            np.isposinf(mu),
+            (x, mu),
+            lambda x, mu: invgamma.logpdf(x, 0.5, scale=0.5),
```

Comment:
I was close to verifying this in my head, but I came up with something slightly different. Can you show how this works out?

Reply:
You can deduce this by looking at equation 5 of the paper and the pdf of the inverse-gamma here: https://en.wikipedia.org/wiki/Inverse-gamma_distribution

Simplification leads to a pdf of an inverse gamma with shape=0.5 and scale=0.5